### PR TITLE
add spinner for AVView, TextView, and mobile ImageView

### PR DESCRIPTION
**Pull request by songgao**
also fix some BarePreview styleing issues

### CODE REVIEW

File: shared/fs/filepreview/bare-preview.native.js [hunk 1]
```diff
@@ -36,65 +36,87 @@ type ConnectedBarePreviewProps = {
...
-    backgroundColor: globalColors.black,
+const styles = Styles.styleSheetCreate({
+  container: Styles.platformStyles({
+    common: {
+      ...Styles.globalStyles.flexBoxColumn,
+      ...Styles.globalStyles.flexGrow,
+      backgroundColor: Styles.globalColors.black,
+    },
+    isIOS: {
+      marginTop: -20, // top status bar
```
Comment (chrisnojima) [inline shared/fs/filepreview/bare-preview.native.js]: i know this is old but you shouldn't need this. you can define the route leaf tags to not be under the status bar

File: shared/fs/filepreview/image-view.native.js [hunk 2]
```diff
@@ -24,30 +28,47 @@ class ImageView extends React.Component<any, {...ImageViewProps, width: number,
...
-          resizeMode="contain" />
+          style={Styles.collapseStyles([
+            styles.image,
+            {
+              height: Math.min(this.state.height, screenHeight),
+              width: Math.min(this.state.width, screenWidth),
+            },
+          ])}
+          onLoadStart={onLoadingStateChange && (() => onLoadingStateChange(true))}
+          onLoadEnd={onLoadingStateChange && (() => onLoadingStateChange(true))}
```
Comment (chrisnojima) [inline shared/fs/filepreview/image-view.native.js]: should be false

File: shared/common-adapters/web-view.desktop.js [hunk 3]
```diff
@@ -23,6 +23,11 @@ class WebView extends React.PureComponent<WebViewProps> {
       ref.insertCSS(css)
       ref.executeJavaScript(javaScript)
     })
+    const {onLoadingStateChange} = this.props
+    if (onLoadingStateChange) {
+      ref.addEventListener('did-start-loading', () => onLoadingStateChange(true))
+      ref.addEventListener('did-stop-loading', () => onLoadingStateChange(false))
```
[the remark this answers is not in the document]
Comment (songgao) [reply shared/common-adapters/web-view.desktop.js]: Hmm I don't see it mentioned in the electron docs https://electronjs.org/docs/api/webview-tag . It does exist for RN webview though.

[the remark this answers is not in the document]
Comment (songgao) [reply shared/common-adapters/web-view.desktop.js]: Just tried it; doesn't seem to work.

File: shared/fs/filepreview/bare-preview.native.js (hunk 1 above)
Comment (chrisnojima) [reply shared/fs/filepreview/bare-preview.native.js]: did you try underStatusbar

Comment (songgao) [reply shared/fs/filepreview/bare-preview.native.js]: Just tried; it worked!

**Pull request merged**